src/main/scala/com/fulcrumgenomics/umi/FilterConsensusReads.scala

```diff
@@ -0,0 +1,217 @@
+    """
+      |Filters a BAM file of consensus reads generated by CallMolecularConsensusReads. Two kinds of
+      |filtering are performed:
+      |  1. Masking/filtering of individual bases in reads
+      |  2. Filtering out of reads
+      |
+      |Base-level filtering of masking is only applied if per-base tags are present. Read-level
+      |filtering is always applied.  When filtering reads, secondary alignments and supplementary
+      |records may be removed independently if they fail one or more filters; if either R1 or R2
+      |primary alignments fail a filter then all records for the template will be filtered out.
+      |
+      |The filters applied are as follows:
+      |  1. Bases with quality scores below min-base-quality are masked to Ns
+      |  2. Bases with fewer than min-reads contributing raw reads are masked to Ns
+      |  3. Bases with a consensus error rate (defined as the fraction of contributing reads that
+      |     voted for a different base than the consensus call) higher than max-base-error-rate
+      |     are masked to Ns
+      |  4. Reads with a proportion of Ns higher than max-no-call-fraction *after* per-base
+      |     filtering are filtered out.
+      |  5. Reads with fewer than min-reads contributing reads are filtered out
+      |  6. Reads with an average consensus error rate higher than max-read-error-rate are
+      |     filtered out
+      |
+      |In order to correctly filter reads in or out by template, if the input BAM is not queryname
+      |sorted or grouped it will be sorted into queryname order.  Likewise, in order to emit
+      |records with corrected NM, UQ and MD tags the reads are re-sorted into coordinate order
+      |for output.
+      |
+      |The reverse-tags-per-base option controls whether per-base tags should be reversed before
+      |being used on reads marked as being mapped to the negative strand.  This is necessary if
+      |the reads have been mapped and the bases/quals reversed but the consensus tags have not.
+      |If true, the tags written to the output BAM will be reversed where necesary in order to
+      |line up with the bases and quals.
+    """
+)
+class FilterConsensusReads
+( @arg(flag="i", doc="The input SAM or BAM file of consensus reads.") val input: PathToBam,
+  @arg(flag="o", doc="Output SAM or BAM file.") val output: PathToBam,
+  @arg(flag="r", doc="Reference fasta file.") val ref: PathToFasta,
+  @arg(flag="R", doc="Reverse [complement] per base tags on reverse strand reads.") val reversePerBaseTags: Boolean = true,
+  @arg(flag="N", doc="Mask (make 'N') consensus bases with quality less than this threshold.") val minBaseQuality: PhredScore = DefaultMinConsensusBaseQuality,
+  @arg(flag="M", doc="The minimum number of reads supporting a consensus base/read.") val minReads: Int = DefaultMinReads,
+  @arg(flag="E", doc="The maximum raw-read error rate across the entire consensus read.") val maxReadErrorRate: Double = 0.025,
```

Comment:
I was actually tempted to go the other way and get rid of the centralized defaults because I think that `CallMolecularConsensusReads` and `FilterConsensusReads` probably want different defaults, with the former being quite low bars (e.g. min=2 q=30) and filter being higher.  What do you think of that?